### ISSUE TITLE
Coding Standards

### ISSUE THREAD

**Issue by jkbe** [HPI-SWA-Teaching/SWT17-Project-05#15]
- give variables/parameters names that let the developer infer the type of the variable by looking at the code without having to debug it
- example:

```smalltalk
printOn: aWriteStream
"instead of"
printOn: aStream
```
- refer to Kent Beck: Smalltak Best Practice Patterns

**Comment by simsieg** [comment]
Commit Messages:

- Write in present tense

- Begin with uppercase letter

Reasons: Unified style, no decision effort

**Comment by simsieg** [comment]
Return style:

```smalltalk
^ returnValue
"instead of"
^returnValue.
```

Reasons: readability, unified style

**Comment by jkbe** [comment]
I don't think that "Commit Messages" falls under the category Coding Standards.
(On the same note: This is a nice guide to good Git commit messages that @FWirtz recommended to me some time back: https://chris.beams.io/posts/git-commit/)

**Comment by simsieg** [comment]
We noticed that our commit messages are not in an unified style and from now we want to be strict about our guidelines. In terms of documentation we commented this here to have a single source of process documentation.

**Comment by flowirtz** [comment]
Really am in favor of the guidelines I recommended to @jak-ing, they discuss several aspects and explain nicely why we should follow these standards.
I do think though that "Commit Messages" are part of Coding Standards in an extended sense, as the Git workflow is the base of our coding work.

**Comment by jkbe** [comment]
Let's discuss this with Jonas on Tuesday, I understand coding standards to be specific to a programming language.
